.github/workflows/ci.yml

```diff
@@ -112,20 +108,29 @@ jobs:
       # Enable corepack for packageManager config
       - run: corepack enable || sudo corepack enable
       - run: yarn install
+      # Set up docker-bake files used by docker/bake-action
+      - run: node_modules/.bin/synthetic-chain prepare-ci
 
-      - name: build test images ${{ matrix.platform }} == ${{ env.DEFAULT_PLATFORM }}
-        run: |
-          docker info
-          node_modules/.bin/synthetic-chain build ${{ matrix.platform == env.DEFAULT_PLATFORM && ' ' || '--dry' }}
-      - name: run test images
-        if: ${{ matrix.platform == env.DEFAULT_PLATFORM }}
+      # Test before pushing the images.
+      - name: run proposal tests
+        if: ${{ matrix.platform == env.X86_PLATFORM }}
         run: node_modules/.bin/synthetic-chain test
 
-      # XXX this should be instant for the local platform because all the stages
-      # were already built in the steps above but it's re-building the last
-      # stage. This is deemed good enough for now.  see
-      # https://github.com/moby/moby/issues/34715
-      - name: Build and push images
+      # Matrix within "bake" instead of GitHub Actions because the actions
```

Comment:
I got confused and thought that this "Matrix" was a platform matrix.  But instead it seems to be the `test,use` matrix.

So the `bake` only produces one platform image per Actions runner?  That's okay for now, but I'd appreciate it if you add a comment to that effect somewhere.